### PR TITLE
feat: Improve casting a vote

### PR DESCRIPTION
**Pull request by slickepinne**
This PR adds/ updates:

**BB app**
- validation for `enqueued messages`

**JS client**
- the option to query `pendingMessages` by `messageId`
- adding one more layer for the cast vote process: now, it checks for the `pendingMessages` to be processed

**BB gem**
- check `pendingMessageStatus`

### CODE REVIEW

**Review by leio10 — APPROVED** [review]
Niiiice!! 👍 😃

File: decidim-bulletin_board-js/src/client/client.js [hunk 1]
```diff
@@ -61,4 +61,16 @@ export class Client {
   processTallyStep(params) {
     return this.apiClient.processTallyStep(params);
   }
+
+  /**
+   * Query PendingMessages for a given messageId
+   *
+   * @param {Object} params - An object that include the following options.
+   *  - {String} messageId - The messageId
+   * @returns {Promise<Object>} - The status of a pendingMessage.
```
Comment (beagleknight) [inline decidim-bulletin_board-js/src/client/client.js:70]: It returns the whole `pendingMessage` now 😄

File: decidim-bulletin_board-js/src/client/graphql-client.js [hunk 2]
```diff
@@ -100,6 +100,25 @@ export class GraphQLClient {
     return result.data.processKeyCeremonyStep.pendingMessage;
   }
 
+  /**
+   * Query PendingMessages for a given messageId
+   *
+   * @param {Object} params - An object that include the following options.
+   *  - {String} messageId - The messageId
+   * @returns {Promise<Object>} - The status of a pendingMessage.
```
Comment (beagleknight) [inline decidim-bulletin_board-js/src/client/graphql-client.js:108]: Same here!

File: decidim-bulletin_board-js/src/voter/voter.js [hunk 3]
```diff
@@ -32,23 +32,43 @@ export class Voter {
...
 
-  async verifyVote(voteHash) {
-    const { id: electionUniqueId } = this.electionContext;
-
-    return new Promise((resolve) => {
+  /**
+   * Confirms if a vote was processed
+   *
+   * @param {Object} messageId - An object that includes the following options.
+   *  - {String} messageId - The messageId
```
Comment (beagleknight) [inline decidim-bulletin_board-js/src/voter/voter.js:39]: I suggest something like "the unique identifier of a message".

File: decidim-bulletin_board-js/src/voter/voter.js [hunk 4]
```diff
@@ -32,23 +32,43 @@ export class Voter {
...
           });
       }, this.options.bulletinBoardWaitTime);
     });
   }
+
+  /**
+   * Verifies a vote
+   *
+   * @param {Object} voteHash - An object that includes the following options.
+   *  - {String} voteHash - the voteHash of a vote
```
Comment (beagleknight) [inline decidim-bulletin_board-js/src/voter/voter.js:63]: You are just giving it the `voteHash` as a `String`, not an object 👍 . Also I recommend renaming it to `contentHash` so you can save some keystrokes.

File: decidim-bulletin_board-ruby/CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+## [0.7.1] - 2021-01-26
```
Comment (beagleknight) [inline decidim-bulletin_board-ruby/CHANGELOG.md:10]: Don't add the version yet. This will be done in the `main` branch before releasing it.

**Review by beagleknight — APPROVED** [review]
Great job! 🤩 . About the artifacts created by the `npm run build` command, don't include them in the PR for now. After this is merged and before the release we will change the package version and recreate the artifacts 👍

File: decidim-bulletin_board-js/src/client/client.js (hunk 1 above)
Comment (slickepinne) [reply decidim-bulletin_board-js/src/client/client.js:70]: 😅

File: decidim-bulletin_board-js/src/voter/voter.js (hunk 4 above)
Comment (slickepinne) [reply decidim-bulletin_board-js/src/voter/voter.js:63]: Great suggestion! Thanks a lot!

**Pull request merged**